ignite/metrics/binary_accuracy.py

```diff
@@ -18,7 +18,7 @@ def reset(self):
 
     def update(self, output):
         y_pred, y = output
-        correct = torch.eq(torch.round(y_pred).type(torch.LongTensor), y)
+        correct = torch.eq(torch.round(y_pred).type(torch.LongTensor), y).view(-1)
         self._num_correct += torch.sum(correct)
         self._num_examples += correct.shape[0]
```

Comment:
I'm not that familiar with evaluating the output of semantic segmentation for example, but now your `num_correct` would be number in range `[0, num_categories*num_pixels]` but your `num_examples` is only adding `batch_size` to the count. Should it not be adding `batch_size*num_categories*num_pixels`?

Reply:
In both cases, now `num_correct` will be a number in range `[0, batch_size * num_pixels]` and `batch_size * num_pixels` will be added to the count because the new shape of `correct` is `(batch_size * num_pixels,)` instead of `(batch_size, image_height, image_width)`. 

Here a gist that shows the differences between the current code and my modifications with simulated data.
https://gist.github.com/chicham/c0128d6e1405246eb99cf0b42731824e
 
Note: num_pixels = image_height * image_width